src/swap/letsexchange.js

```diff
@@ -0,0 +1,294 @@
+
+      if (networkFrom !== fromCurrencyCode) {
+        networkFrom = networkCodeConverter(networkFrom);
+      }
+
+      if (networkTo !== toCurrencyCode) {
+        networkTo = networkCodeConverter(networkTo);
+      }
+
+      // Convert the native amount to a denomination:
+      const quoteAmount =
+        request.quoteFor === 'from'
+          ? await request.fromWallet.nativeToDenomination(
+              request.nativeAmount,
+              request.fromCurrencyCode
+            )
+          : await request.toWallet.nativeToDenomination(
+              request.nativeAmount,
+              request.toCurrencyCode
+            )
+
+      // Swap the currencies if we need a reverse quote:
+      const quoteParams = {
+        from: fromCurrencyCode,
+        to: toCurrencyCode,
+        network_from: networkFrom,
+        network_to: networkTo,
+        amount: quoteAmount
+      }
+      log('quoteParams:', quoteParams)
+
+      // Calculate the amounts:
+      let fromAmount, fromNativeAmount, toNativeAmount, reply
+      if (request.quoteFor === 'from') {
+        reply = await call(uri + 'info', request, {
+          params: quoteParams
+        })
+        fromAmount = quoteAmount
+        fromNativeAmount = request.nativeAmount
+        toNativeAmount = await request.toWallet.denominationToNative(
+          reply.amount.toString(),
+          request.toCurrencyCode
+        )
+      } else {
+        reply = await call(uri + 'info-revert', request, {
+          params: quoteParams
+        })
+        fromAmount = reply.amount
+        fromNativeAmount = await request.fromWallet.denominationToNative(
+          fromAmount.toString(),
+          request.fromCurrencyCode
+        )
+        toNativeAmount = request.nativeAmount
+      }
+      log('fromNativeAmount' + fromNativeAmount)
+      log('toNativeAmount' + toNativeAmount)
+
+      // Check the minimum:
+      const nativeMin = await request.fromWallet.denominationToNative(
+        reply.min_amount,
+        request.fromCurrencyCode
+      )
+      if (lt(fromNativeAmount, nativeMin)) {
+        throw new SwapBelowLimitError(swapInfo, nativeMin)
+      }
+
+      const { promoCode } = opts
+      const sendReply = await call(uri + 'transaction', request, {
+        params: {
+          deposit_amount: fromAmount,
+          coin_from: fromCurrencyCode,
+          coin_to: toCurrencyCode,
+          network_from: networkFrom,
+          network_to: networkTo,
+          withdrawal: toAddress,
+          return: fromAddress,
+          // return_extra_id: 'empty',
+          // withdrawal_extra_id: 'empty',
+          return_extra_id: null,
+          withdrawal_extra_id: null,
+          affiliate_id: initOptions.apiKey,
+          promocode: promoCode != null ? promoCode : '',
+          type: 'edge',
+          float: true,
+          isEstimate: false
+        }
+      })
+      log('sendReply' + sendReply)
+      const quoteInfo: QuoteInfo = sendReply
```

Comment:
This should use a `cleaner`. See https://github.com/EdgeApp/edge-exchange-plugins/blob/master/src/swap/sideshift.js#L377

Reply:
Is there an example?

Reply:
The link has an example

Reply:
Fixed it.